### PR TITLE
--disable-asm is required to compile on FreeBSD

### PR DESCRIPTION
**Pull request by dbolser**
Adding a note about the discussion here for compiling on FreeBSD:
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=226155

### CODE REVIEW

File: doc/build-unix.md [hunk 1]
```diff
@@ -328,8 +328,11 @@ Then build using:
     ./autogen.sh
     ./configure --disable-wallet # OR
     ./configure BDB_CFLAGS="-I${BDB_PREFIX}/include" BDB_LIBS="-L${BDB_PREFIX}/lib -ldb_cxx"
+    
     gmake
 
+*Note* that as of Bitcoin Core version 0.16.0, you need to add --disable-asm at the configure step to disable defailt ASM optimization to compile for FreeBSD. See [here](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=226155) for details.
```
Comment (fanquake) [inline doc/build-unix.md]: s/defailt/default

**Comment by dbolser** [comment]
Thanks. Should I fix that?​

File: doc/build-unix.md (hunk 1 above)
Comment (practicalswift) [inline doc/build-unix.md]: Nit: Use \`--disable-asm\` to get correct formatting :-)

**Comment by laanwj** [comment]
I cannot reproduce this. What version of FreeBSD?

**Comment by dbolser** [comment]
The ticket I've cited says 10.3, 10.4, 11.1 and 12C. I don't actually know the impact of disabling ASM, or if there is a better fix than the one suggested by the port maintainers.

File: doc/build-unix.md (hunk 1 above)
Comment (dbolser) [reply doc/build-unix.md]: Done

Comment (dbolser) [reply doc/build-unix.md]: Done.

**Comment by dbolser** [comment]
Sorry, just spotted your edits. I'll try building master  when I get home tonight (this is an issue for me on the 0.16 branch). Cheers,

**Comment by laanwj** [comment]
Yes, no matter what, I think we should solve this in the build system (by making asm detection more robust), not burden the user through the documentation.

**Comment by dbolser** [comment]
Yes, good point. It didn't work for me on master BTW (same error). I wonder
why you're not seeing it.

**Comment by dbolser** [comment]
OK, I'll try it at home later toady.

**Comment by laanwj** [comment]
FWIW here's my config.log:

https://0bin.net/paste/Mm9HmzaWU7YOhbF5#ySh1eN90wjJDDJXNW-ktQ0OOjeBgRfqmRKUF38jfNmZ

**Comment by theuni** [comment]
I managed to reproduce with an old clang from: http://releases.llvm.org/3.4.2/clang+llvm-3.4.2-x86_64-linux-gnu-ubuntu-14.04.xz

I can confirm that the above patch fixes the issue, although another arises. See here for the next fix: https://github.com/theuni/bitcoin/commit/8f097ef8b9a1d36b1aa82e6428da3884fb9eebf8

With those two patches, I'm able to compile the .cpp without issues, and the sha2 tests pass. If @dbolser confirms that it's fixed for FreeBSD, I'll PR that branch.

**Comment by laanwj** [comment]
Not sure what to do here. If you cannot reproduce the problem anymore, might want to close.

**Comment by theuni** [comment]
Agreed. Ideally we'd compile with any c++11-compatible compiler, but given that this hits 2 separate clang bugs, it's probably best to just require a non-broken version.

**Pull request closed without merging**